mogc_partnerships/permissions.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework.permissions import BasePermission
+
+from .lib import get_cohort
+
+
+class ManagerCreatePermission(BasePermission):
+    managed_methods = "POST"
+
+    def has_permission(self, request, view):
+        if request.method not in self.managed_methods:
+            return True
+
+        user = request.user
+        serializer = view.get_serializer(data=request.data)
+        if not serializer.is_valid():
+            return False
+
+        partner = serializer.validated_data["partner"]
+        if user in partner.managers.all():
+            return True
+
+        return False
+
+
+class ManagerEditPermission(BasePermission):
+    managed_methods = ("PUT", "PATCH")
+
+    def has_permission(self, request, view):
+        if request.method not in self.managed_methods:
+            return True
+
+        user = request.user
+        cohort = get_cohort(request.user, view.kwargs.get("cohort_uuid"))
```

Comment:
Same as above, we'll end up doing this multiple times - once in the permission, and again in the view method, which isn't ideal.

Reply:
Could attach this to the session in a pre-request hook perhaps.